onfido-java/src/test/java/com/onfido/integration/DocumentManagerTest.java

```diff
@@ -6,164 +6,126 @@
+  @BeforeMethod
+  public void setupTest() throws Exception {
+    applicant = createApplicant();
+    document = uploadDocument(applicant, "file.png");
+  }
 
-    InputStream inputStream = new ByteArrayInputStream("testing testing 1 2".getBytes());
+  @Test
+  public void uploadDocumentTest() throws Exception {
+    assertRequestBodyContains("applicant_id");
+    assertRequestBodyContains(applicant.getId());
+    assertRequestBodyContains("issuing_country");
+    assertRequestBodyContains("USA");
+    assertRequestBodyContains("side");
+    assertRequestBodyContains("front");
+    assertRequestBodyContains("type");
+    assertRequestBodyContains("passport");
+    assertRequestBodyContains("{\"country_of_residence\":\"GBR\",\"ip_address\":\"127.0.0.1\"}");
 
-    Location.Request locationRequest =
-        Location.request()
-            .ipAddress("127.0.0.1")
-            .countryOfResidence("GBR");
-
-    Document.Request documentRequest =
-        Document.request()
-            .applicantId("test id")
-            .issuingCountry("USA")
-            .side("front")
-            .type("passport")
-            .location(locationRequest);
-
-    Document document = onfido.document.upload(inputStream, "file.png", documentRequest);
-
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/documents/", request.getPath());
-
-    // Correct request body
-    String requestBody = request.getBody().readUtf8();
-    Assert.assertTrue(requestBody.contains("applicant_id"));
-    Assert.assertTrue(requestBody.contains("test id"));
-    Assert.assertTrue(requestBody.contains("issuing_country"));
-    Assert.assertTrue(requestBody.contains("USA"));
-    Assert.assertTrue(requestBody.contains("side"));
-    Assert.assertTrue(requestBody.contains("front"));
-    Assert.assertTrue(requestBody.contains("type"));
-    Assert.assertTrue(requestBody.contains("passport"));
-    Assert.assertTrue(requestBody.contains("testing testing 1 2"));
-    Assert.assertTrue(requestBody.contains("{\"country_of_residence\":\"GBR\",\"ip_address\":\"127.0.0.1\"}"));
-
-    // Correct response body
     assertEquals("file.png", document.getFileName());
   }
 
   @Test
-  public void downloadDocument() throws Exception {
-    MockWebServer server = mockFileRequestResponse("test", "image/png");
+  public void downloadDocumentTest() throws Exception {
+    prepareMock("test", "image/png", 0);
 
-    Onfido onfido =
-        Onfido.builder().apiToken("token").unknownApiUrl(server.url("/").toString()).build();
+    FileDownload download = onfido.document.download(document.getId());
 
-    FileDownload download = onfido.document.download("document_id");
+    takeRequest("/documents/" + document.getId() + "/download");
 
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/documents/document_id/download", request.getPath());
+    if (isMockingEnabled()) {
```

Comment:
Same suggestion as above on skipping tests. Also, the code style here is messed around in the `if/else` 💄 😝